conformance_tests/tools/metrics/src/test_metric_tracer.cpp

```diff
@@ -119,6 +120,590 @@ TEST_F(
+                              &metric_tracer_handle);
+    lzt::metric_tracer_enable(metric_tracer_handle, true);
+    LOG_DEBUG << "execute workload";
+    lzt::execute_command_lists(command_queue, 1, &command_list, nullptr);
+    LOG_DEBUG << "synchronize with completion of workload";
+    lzt::synchronize(command_queue, std::numeric_limits<uint64_t>::max());
+
+    size_t raw_data_size{};
+    std::vector<uint8_t> raw_data;
+    uint64_t raw_data_accumulate{};
+    result = zetMetricTracerReadDataExp(metric_tracer_handle, &raw_data_size,
+                                        nullptr);
+    ASSERT_EQ(ZE_RESULT_SUCCESS, result)
+        << "zetMetricTracerReadDataExp failed while retrieving the raw data "
+           "size";
+    ASSERT_NE(0, raw_data_size) << "zetMetricTracerReadDataExp reports that "
+                                   "there are no metrics available to read";
+    raw_data.resize(raw_data_size, 0);
+    result = zetMetricTracerReadDataExp(metric_tracer_handle, &raw_data_size,
+                                        raw_data.data());
+    ASSERT_EQ(ZE_RESULT_SUCCESS, result)
+        << "zetMetricTracerReadDataExp failed while retrieving the raw data";
+
+    raw_data_accumulate =
+        std::accumulate(raw_data.begin(), raw_data.end(), 0ULL);
+    ASSERT_NE(0, raw_data_accumulate)
+        << "all raw data entries are zero, zetMetricTracerReadDataExp is "
+           "expected to read useful data";
+
+    LOG_DEBUG << "execute workload";
+    lzt::execute_command_lists(command_queue, 1, &command_list, nullptr);
+    LOG_DEBUG << "synchronize with completion of workload";
+    lzt::synchronize(command_queue, std::numeric_limits<uint64_t>::max());
+
+    lzt::metric_tracer_disable(metric_tracer_handle, false);
+
+    const char *value_string =
+        std::getenv("LZT_METRIC_DISABLE_TRACER_MAX_WAIT_TIME");
+    uint32_t max_wait_time_in_milliseconds = 10;
+    if (value_string != nullptr) {
+      uint32_t value = atoi(value_string);
+      max_wait_time_in_milliseconds =
+          value != 0 ? value : max_wait_time_in_milliseconds;
+      max_wait_time_in_milliseconds =
+          std::min(max_wait_time_in_milliseconds, 100u);
+    }
+    /* read raw data while API returns ZE_RESULT_SUCCESS and
+     * ZE_RESULT_NOT_READY(for the last piece of data) */
+    do {
+      raw_data_size = 0;
+      result = zetMetricTracerReadDataExp(metric_tracer_handle, &raw_data_size,
+                                          nullptr);
+      EXPECT_TRUE(ZE_RESULT_SUCCESS == result || ZE_RESULT_NOT_READY == result)
+          << "zetMetricTracerReadDataExp failed while retrieving the raw data "
+             "size";
+
+      if (raw_data_size != 0) {
+        raw_data.resize(raw_data_size, 0);
+        result = zetMetricTracerReadDataExp(metric_tracer_handle,
+                                            &raw_data_size, raw_data.data());
+        EXPECT_TRUE(ZE_RESULT_SUCCESS == result ||
```

Comment:
I think you had a previous version of this test that was checking the read size in the loop: if got success, retrieve the data and retry; if got NOT READY, exit the loop and retrieve the data, confirm data is more than zero.  Then read query size again and confirm get not ready and size 0. 
This is more accurately checking the expected behavior.

Reply:
clarified on the phone. No changes needed.